kettle-plugins/formats-meta/src/main/java/org/pentaho/big/data/kettle/plugins/formats/avro/output/AvroOutputMetaBase.java

```diff
@@ -286,7 +286,12 @@ public void saveRep( Repository rep, IMetaStore metaStore, ObjectId transformati
         rep.saveStepAttribute( transformationId, stepId, i, PRECISION, field.getPrecision() );
         rep.saveStepAttribute( transformationId, stepId, i, SCALE, field.getScale() );
         rep.saveStepAttribute( transformationId, stepId, i, NULLABLE, Boolean.toString( field.getAllowNull() ) );
-        rep.saveStepAttribute( transformationId, stepId, i, DEFAULT, field.getDefaultValue() );
+        String defaultValue = field.getDefaultValue();
+        int length = defaultValue.length();
+        if(length>0)
+          rep.saveStepAttribute( transformationId, stepId, i, DEFAULT, defaultValue );
+        else
+          rep.saveStepAttribute( transformationId, stepId, i, DEFAULT, null );
```

Comment:
I'd like to see this written in a way where the rep.saveStepAttribute(...) method did not have to be repeated. Something like this:
        String defaultValue = StringUtil.isEmpty( field.getDefaultValue() ) ? null : field.getDefaultValue();
        rep.saveStepAttribute( transformationId, stepId, i, DEFAULT, defaultValue );